### PR TITLE
[Re-submit] Compute true loss Flax examples

### PR DESCRIPTION
**Pull request by duongna21**
Re-submit #18458.

cc @patrickvonplaten @sanchit-gandhi

### CODE REVIEW

File: examples/flax/language-modeling/run_bart_dlm_flax.py [hunk 1]
```diff
@@ -888,7 +894,7 @@ def eval_step(params, batch):
                 num_eval_samples = len(tokenized_datasets["validation"])
                 # Avoid using jax.numpy here in case of TPU training
                 eval_samples_idx = np.arange(num_eval_samples)
-                eval_batch_idx = generate_batch_splits(eval_samples_idx, eval_batch_size, drop_last=False)
+                eval_batch_idx = generate_batch_splits(eval_samples_idx, eval_batch_size)
```
Comment (sanchit-gandhi) [inline examples/flax/language-modeling/run_bart_dlm_flax.py:897]: Is `drop_last=False` causing issues with `nan`? If so feel free to open a new issue and we can discuss a solution!

**Review by sanchit-gandhi — APPROVED** [review]
Thanks for re-submitting the PR @duongna21!

Here's a tip for pushing after a rebase that'll hide all the commits made by others in the diff of your PR...

Generally, we use the tracking arg when pushing to the remote repository:
```
git push -u true-loss-flax-examples
```

Unfortunately, GitHub deals very badly with rebases - it may show all the files touched by the rebase in the diff of your PR (as was the case in #18458!). To avoid this, it’s best to force-push after the rebase:
```
git push -f true-loss-flax-examples
```

This is the only time we tend to force-push. Hope that helps in explaining how to push after a rebase and makes submitting new PRs all the easier! I'm explaining this as I've made exactly the same mistake before 😅

File: examples/flax/language-modeling/run_bart_dlm_flax.py (hunk 1 above)
Comment (duongna21) [reply examples/flax/language-modeling/run_bart_dlm_flax.py:897]: @sanchit-gandhi Nice tip! Thank you very much.
